src/model/points-model.js

```diff
@@ -20,20 +24,44 @@ export default class PointsModel extends Observable {
     return this.#points;
   }
 
-  updatePoint = (updateType, update) => {
+  init = async () => {
+    try {
+      const points = await this.#pointApiService.points;
+      const offers = await this.#pointApiService.offers;
+      const destinations = await this.#pointApiService.destinations;
+
+      this.#points = points.map(this.#adaptToClient);
+      this.#offers = offers;
+      this.#destinations = destinations;
+    } catch(err) {
+      this.#points = [];
+      this.#offers = [];
+      this.#destinations = [];
+    }
+
+    this._notify(UpdateType.INIT);
+  };
+
+  updatePoint = async (updateType, update) => {
     const index = this.#points.findIndex((point) => point.id === update.id);
 
     if (index === -1) {
       throw new Error(`Can't delete unexisting point: ${update}`);
     }
 
-    this.#points = [
-      ...this.#points.slice(0, index),
-      update,
-      ...this.#points.slice(index + 1)
-    ];
+    try {
+      const response = await this.#pointApiService.updatePoint(update);
+      const updatedPoint = this.#adaptToClient(response);
 
-    this._notify(updateType, update);
+      this.#points = [
+        ...this.#points.slice(0, index),
+        updatedPoint,
+        ...this.#points.slice(index + 1)
+      ];
+      this._notify(updateType, updatedPoint);
+    } catch(err) {
+      throw new Error('Can\'t update point');
```

Comment:
Всегда надо стараться ошибку выкидывать наиболее информативно чтобы твои коллеги быстрее понимали в чем проблема. 
```suggestion
      throw new Error(`Can't update point: ${update}. Error: ${err}`);
```